### ISSUE TITLE
Kernel Panic when sharing screen in meeting calls from browsers (Only Firefox tested)

### ISSUE THREAD

**Issue by Virgula0** [hyprwm/Hyprland#2378]
This is a pretty annoying bug which force me to do an hard shutdown of my computer when I try to share my screen from a web browser via discord or other meeting call apps.
Let me know what info are useless to debug the problem and I will provide the output of needed commands.

My system:

```
❯ neofetch
                   -`                    angelo@virgularch 
                  .o+`                   ----------------- 
                 `ooo/                   OS: Arch Linux x86_64 
                `+oooo:                  Host: 21CF000BUS ThinkPad T14 Gen 3 
               `+oooooo:                 Kernel: 6.3.2-arch1-1 
               -+oooooo+:                Uptime: 7 mins 
             `/:-:++oooo+:               Packages: 742 (pacman) 
            `/++++/+++++++:              Shell: bash 5.1.16 
           `/++++++++++++++:             Resolution: 1920x1200 
          `/+++ooooooooooooo/`           DE: Hyprland 
         ./ooosssso++osssssso+`          Theme: Adwaita-dark [GTK2/3] 
        .oossssso-````/ossssss+`         Icons: Adwaita [GTK2/3] 
       -osssssso.      :ssssssso.        Terminal: kitty 
      :osssssss/        osssso+++.       CPU: AMD Ryzen 5 PRO 6650U with Radeon Graphics (12) @ 2.900GHz 
     /ossssssss/        +ssssooo/-       GPU: AMD ATI Radeon 680M 
   `/ossssso+/:-        -:/+osssso+-     Memory: 1551MiB / 14746MiB 
  `+sso+:-`                 `.-/+oso:
 `++:.                           `-/+/                           
 .`                                 `/                           

```

I need to check if this happens with chromium too.
Firefox version: 113.0.1 (64-bit)
I found also a bug crash report into the hyprland logs but I don't know if it exactly refers to the crash that I'm reporting. Here is the log:

```
--------------------------------------------
   Hyprland Crash Report
--------------------------------------------
I don't feel so good...

Hyprland received signal 11 (Segmentation fault)

Version: f27873a6f06dc2f87600edb890f3c38298bfb55f

System info:
	System name: Linux
	Node name: virgularch
	Release: 6.3.2-arch1-1
	Version: #1 SMP PREEMPT_DYNAMIC Thu, 11 May 2023 16:40:42 +0000

GPU:
	04:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt [Radeon 680M] [1002:1681] (rev d2) (prog-if 00 [VGA controller])


os-release:
	NAME="Arch Linux"
	PRETTY_NAME="Arch Linux"
	ID=arch
	BUILD_ID=rolling
	ANSI_COLOR="38;2;23;147;209"
	HOME_URL="https://archlinux.org/"
	DOCUMENTATION_URL="https://wiki.archlinux.org/"
	SUPPORT_URL="https://bbs.archlinux.org/"
	BUG_REPORT_URL="https://bugs.archlinux.org/"
	PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
	LOGO=archlinux-logo
	


Backtrace:
	#0 | Hyprland(_ZN13CrashReporter18createAndSaveCrashEi+0x450) [0x5617ef6a5b20]
		??
		??:0
	#1 | Hyprland(_Z25handleUnrecoverableSignali+0x47) [0x5617ef646667]
		??
		??:0
	#2 | /usr/lib/libc.so.6(+0x39ab0) [0x7fa4a9b62ab0]
		??
		??:0
	#3 | Hyprland(_ZN12CFramebuffer5allocEii+0x253) [0x5617ef71c4e3]
		??
		??:0
	#4 | Hyprland(_ZN15CHyprOpenGLImpl18makeWindowSnapshotEP7CWindow+0x25b) [0x5617ef729b8b]
		??
		??:0
	#5 | Hyprland(_ZN6Events20listener_unmapWindowEPvS0_+0x252) [0x5617ef6c1e12]
		??
		??:0
	#6 | Hyprland(_Z13handleWrappedP11wl_listenerPv+0x40) [0x5617ef6d0e30]
		??
		??:0
	#7 | /usr/lib/libwayland-server.so.0(wl_signal_emit_mutable+0x7e) [0x7fa4aa4ed01e]
		??
		??:0
	#8 | /usr/lib/libwlroots.so.12032(+0x9b318) [0x7fa4aa5d1318]
		??
		??:0
	#9 | /usr/lib/libwlroots.so.12032(+0x9c2f0) [0x7fa4aa5d22f0]
		??
		??:0
	#10 | /usr/lib/libwlroots.so.12032(+0xa6610) [0x7fa4aa5dc610]
		??
		??:0
	#11 | /usr/lib/libwayland-server.so.0(wl_event_loop_dispatch+0xa2) [0x7fa4aa4eeae2]
		??
		??:0
	#12 | /usr/lib/libwayland-server.so.0(wl_display_run+0x27) [0x7fa4aa4ef2d7]
		??
		??:0
	#13 | Hyprland(main+0x4a7) [0x5617ef635e57]
		??
		??:0
	#14 | /usr/lib/libc.so.6(+0x23850) [0x7fa4a9b4c850]
		??
		??:0
	#15 | /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7fa4a9b4c90a]
		??
		??:0
	#16 | Hyprland(_start+0x25) [0x5617ef643615]
		??
		??:0


Log tail:
[LOG] Callback 5617f1b060a8 -> 5617f1b060a0, XDG Window Late removed.
[LOG] Callback 5617f1b06248 -> 5617f1b06240, XDG Window Late removed.
[LOG] Callback 5617f1b06380 -> 5617f1b06378, XDG Window Late removed.
[LOG] Callback 5617f1b064b8 -> 5617f1b064b0, XDG Window Late removed.
[LOG] Callback 5617f1b06450 -> 5617f1b06448, XDG Window Late removed.
[LOG] Callback 5617f1b063e8 -> 5617f1b063e0, XDG Window Late removed.
[LOG] Callback 5617f1b06520 -> 5617f1b06518, XDG Window Late removed.
[LOG] Callback 5617f1b06318 -> 5617f1b06310, XDG Window Late removed.
[LOG] Framebuffer created, status 36053
[LOG] Removing last node (dwindle)
[LOG] On closed window, new focused candidate is 0
[LOG] Destroying the SubSurface tree of unmapped window 5617f1b06080
[LOG] Callback 5617f1aa6208 -> 5617f1aa6200, SurfaceTreeNode removed.
[LOG] Callback 5617f1aa6270 -> 5617f1aa6268, SurfaceTreeNode removed.
[LOG] Callback 5617f1aa61a0 -> 5617f1aa6198, SurfaceTreeNode removed.
[LOG] Callback 5617f1aa6138 -> 5617f1aa6130, CWLSurface removed.
[LOG] CWLSurface 5617f1aa6108 called destroy()
[LOG] SurfaceTree Node removed
[LOG] Callback 5617f1a99a90 -> 5617f1a99a88, Subsurface removed.
[LOG] Callback 5617f1a999c0 -> 5617f1a999b8, Subsurface removed.
[LOG] Callback 5617f1a99a28 -> 5617f1a99a20, Subsurface removed.
[LOG] Callback 5617f1aa6ed8 -> 5617f1aa6ed0, SurfaceTreeNode removed.
[LOG] Callback 5617f1aa6f40 -> 5617f1aa6f38, SurfaceTreeNode removed.
[LOG] Callback 5617f1aa6e70 -> 5617f1aa6e68, SurfaceTreeNode removed.
[LOG] SurfaceTree Node removed
[LOG] Callback 5617f1b066c0 -> 5617f1b066b8, Toplevel removed.
[LOG] Callback 5617f1b06658 -> 5617f1b06650, Toplevel removed.
[LOG] Callback 5617f1b06728 -> 5617f1b06720, Toplevel removed.
[LOG] Callback 5617f1b06800 -> 5617f1b067f8, CWLSurface removed.
[LOG] CWLSurface 5617f1b067d0 called destroy()
[LOG] Callback 5617f1b06178 -> 5617f1b06170, CWindow removed.
[LOG] Window 5617f1b06080 destroyed, queueing. (class )
[LOG] Callback 5617f1b06110 -> 5617f1b06108, XDG Window removed.
[LOG] Callback 5617f1b061e0 -> 5617f1b061d8, XDG Window removed.
[ERR] onWindowRemovedTiling node null?
[LOG] Callback 5617f1b52b98 -> 5617f1b52b90, IdleInhibitor removed.
[LOG] Destroyed an idleinhibitor
[LOG] Window 5617f1b4a450 unmapped (class discord)
[LOG] Unregistered late callbacks XWL
[LOG] Callback 5617f1b4a6e8 -> 5617f1b4a6e0, XWayland Window Late removed.
[LOG] Callback 5617f1b4a958 -> 5617f1b4a950, XWayland Window Late removed.
[LOG] Callback 5617f1b4a618 -> 5617f1b4a610, XWayland Window Late removed.
[LOG] Callback 5617f1b4a820 -> 5617f1b4a818, Xwayland Window Late removed.
[CRITICAL] 
==========================================================================================
ASSERTION FAILED! 

Framebuffer incomplete, couldn't create! (FB status: 0)

at: line 43 in Framebuffer.cpp

```

Config for hyprland used: https://github.com/SolDoesTech/HyprV4
Original issue thread: https://github.com/SolDoesTech/HyprV4/issues/14#issue-1728044655

**Comment by vaxerski** [comment]
looks like a driver issue?

**Comment by romanstingler** [comment]
Have you tried upgrading FF? 
could be related to 
https://github.com/emersion/xdg-desktop-portal-wlr/issues/271#issuecomment-1562721238

**Comment by Virgula0** [comment]
Is not a browser problem, it happens with chromium too, it's wayland and it's neither a kernel panic I guess

**Comment by romanstingler** [comment]
have you resolved it?
Have you tried installing firefox-nightly from AUR and testing it there? 

Otherwise, I would suggest deleting all the configs (or move to another DIR) and trying to install a plain version of hyprland.

**Comment by romanstingler** [comment]
@Virgula0 could you resolve the issue??

**Comment by Virgula0** [comment]
> @Virgula0 could you resolve the issue??

No, screen sharing on Firefox-nightly doesn't work. Don't know what to do to solve it

**Comment by samflores** [comment]
I have the same issue. I'm able to share my screen, but eventually Hyprland crashes with a similar stack trace. Sometimes is happens a few minutes after I start sharing, sometimes it takes hours. It already happened when sharing using Firefox, Chrome and Qutebrowser (which uses Webkit).
One thing that I just noticed is that my laptop has the same GPU as @Virgula0's - `Advanced Micro Devices, Inc. [AMD/ATI] Rembrandt [Radeon 680M] [1002:1681]`. If fact, my laptop also has a second, dedicated, GPU (also an AMD Radeon). I'll try to force the browser to use the dedicated GPU and report back if the crash still occurs.

**Comment by samflores** [comment]
Since yesterday I've shared my screen for more that 7 hours and no crashes happened. I used `switcherooctl launch -g 1 google-chrome-stable` to force it to run with the dedicated GPU and used `radeontop` to monitor the usage of both GPUs and confirmed Chrome was using the dedicated one while screen sharing (it does not use it when started normally). I don't have a clue on how to actually fix it to run using the integrated GPU, but at least this suggests it's a driver issue and I found a workaround for it.
